src/lib/components/VideoCard/VideoCard.svelte

```diff
@@ -17,17 +34,44 @@
       console.warn("Unsupported video type; could not embed video.");
     }
   }
+
+  afterUpdate(async () => {
+    if (youtubeVideoId && (!title || !description)) {
+      const snippet = await getYoutubeVideoData(youtubeVideoId);
+      title = snippet.title;
+      description = snippet.description;
+    }
+  });
 </script>
 
-<!-- TODO: Build out card instead of just embedding video directly in page. -->
-<!-- TODO: Make embed responsive. -->
 {#if youtubeVideoId}
-  <iframe
-    width="560"
-    height="315"
-    src={`https://www.youtube.com/embed/${youtubeVideoId}`}
-    title="Embedded YouTube video"
-    frameborder="0"
-    allowfullscreen
-  />
+  {#key youtubeVideoId}
```

Comment:
Mentioned this before, if we don't include this then the "subscribe" button below gets re-added for some reason. There's probably a better way to handle this so that both the `afterUpdate` and the `key` aren't needed.